app/lib/cashu/types.ts

```diff
@@ -104,6 +88,32 @@ export type NUT10Secret = {
   tags?: NUT10SecretTag[];
 };
 
+export const RawNUT10SecretSchema = z.tuple([
+  z.enum(WELL_KNOWN_SECRET_KINDS),
+  z.object({
+    nonce: z.string(),
+    data: z.string(),
+    tags: z.array(NUT10SecretTagSchema).optional(),
+  }),
+]) satisfies z.ZodType<[WellKnownSecretKind, Omit<NUT10Secret, 'kind'>]>;
```

Comment:
do we need this satisfies clause?

Reply:
Not necessary. Why would you remove it? I like it because it it helps me know I wrote the schema correctly, but if its not written correctly, then the tests should reveal that pretty quickly

Reply:
please see https://github.com/MakePrisms/boardwalkcash/pull/233/commits/9dd6ae9f144fac83ffeb184162c128d84369ee3d

I defined NUT10Secret and WellKnownSecretKind as schemas, then created the RawNut10Secret schema by composing those two new schemas

Reply:
looks good